### PR TITLE
Re-enable and rewrite parts of menu e2etest

### PR DESCRIPTION
**Pull request by lawrencewin**
E2E testing for the Menu component was disabled until now due to the spec failing in the pipeline. 

This PR re-enables the Menu tests and fixes the logic errors within the test spec which caused failures in CI. 

### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

- Removed the spec file from the list of excluded specs in the wdio config.
- Use getAttribute('ExpandCollapse.ExpandCollapseState') on primary component to check if menu is open rather than the value of a text component controlled by the menu

### Verification

2 / 3 functional tests were failing before. Now they all pass.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts [hunk 1]
```diff
@@ -23,25 +22,19 @@ class MenuPageObject extends BasePage {
...
-    await browser.waitUntil(async () => await callbackText.isDisplayed(), {
-      timeout: this.waitForUiEvent,
-      timeoutMsg: 'The Menu did not close.',
-      interval: 1000,
-    });
-
-    return await callbackText.isDisplayed();
+  async menuIsExpanded(): Promise<boolean> {
+    const expandState = await (await this._primaryComponent).getAttribute('ExpandCollapse.ExpandCollapseState');
+    console.log(expandState);
```
Comment (PPatBoyd) [inline apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:36]: would it be fair to say that we're testing that the ExpandCollapse state changes as-expected when we invoke the button, but we haven't actually verified that a menu appeared on-screen? Asking in part relative to the commented out code in Menu.spec.win.ts re: `TestID for Callouts aren't propagating, so this test is failing`

Comment (lawrencewin) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts]: This is a fair assessment. I've added a line of code that also tests if the primary menu item is also displayed in the same method.

Comment (samuelfreiberg) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts]: The accessibility tests also open the menu and query for a menu item, so we verify the menu appears there. And the line of code Larry added does the same.

@lawrencewin Perhaps we create a new accessibility test to check for the ExpandCollapse state (since it is the accessibility state). And then in the functional testing block, we only check if the menu is open by determining if the menu item is visible or not.

This way, we don't have 1 test checking for two separate conditions (accessibility state AND menu item visibility).

We want these tests to be as specific as possible. Right now, if this test fails, the developer has no idea if it failed because the ExpandCollapse state was wrong or if the menu item wasn't displayed.

Does that make sense?

File: apps/fluent-tester/src/TestComponents/Menu/consts.ts [hunk 2]
```diff
@@ -16,3 +16,8 @@ export const MENU_ON_OPEN = 'Menu_On_Open'; // For testing the open functionalit
 export const MENU_ON_CLOSE = 'Menu_On_Close'; // For testing the open functionality of Menu
 
 export const MENU_DEFOCUS_BUTTON = 'Menu_Defocus_Button'; // Used to move focus away from menu button
+
+export const enum ExpandCollapseState {
```
Comment (PPatBoyd) [inline apps/fluent-tester/src/TestComponents/Menu/consts.ts:20]: Would it make sense to share the ExpandCollapseState values between all test components?

File: apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts [hunk 3]
```diff
@@ -78,6 +74,10 @@ class MenuPageObject extends BasePage {
     return await this._primaryComponent;
```
Comment (PPatBoyd) [inline apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:74]: driveby comment -- if this function always returns `await this._primaryComponent`, can we simplify or correct its purpose/add a use case?

**Review by PPatBoyd — APPROVED** [review]

File: apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts (hunk 1 above)
Comment (PPatBoyd) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts]: @samuelfreiberg ahh yeah my comment was mostly directed at clarifying this function vs the test as a whole, I articulated that poorly.  I'm not totally sure of all interpretations I made while reading this PR, can you help clarifying for me for me how the `'Validate OnOpenChange() callback was fired -> Type "SPACE"'` test queries the menu item?

File: apps/fluent-tester/src/TestComponents/Menu/consts.ts (hunk 2 above)
Comment (samuelfreiberg) [reply apps/fluent-tester/src/TestComponents/Menu/consts.ts:20]: Agreed with Pat. This should probably go in common/consts

File: apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts [hunk 4]
```diff
@@ -78,6 +74,10 @@ class MenuPageObject extends BasePage {
...
   }
 
+  async resetTest() {
+    await (await this._defocusButton).click();
```
Comment (samuelfreiberg) [inline apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:78]: Does this actually close the Menu? I just tried on the tester app and it didn't. Can you confirm?

**Review by samuelfreiberg — CHANGES_REQUESTED** [review]

File: apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts (hunk 4 above)
Comment (PPatBoyd) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:78]: it should light dismiss off a click (testable :D) but also `escape` is a surefire menu dismissal

Comment (samuelfreiberg) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:78]: What is a "light" dismiss?

Comment (PPatBoyd) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts:78]: @lawrencewin fyi for context; `light dismiss UI` is transient UI such as callouts and menus that have less screen presence/permanence than dialogs (transient UI that requires explicit user dismissal).  The most common action is a tap/click away from the light-dismiss UI, but ranges in behavior between features and apps. [Our Callout on Win32 currently offers](https://github.com/microsoft/fluentui-react-native/blob/6dea077f3b2fc93fc533128ecbc58d00cc9644ed/packages/components/Callout/src/Callout.types.ts#L29) `preventDismissOnKeyDown` and `preventDismissOnClickOutside` to disable these common light-dismiss interactions, which could be used in UI such as an @ mention picker or a teaching UI bubble (respectively) to allow them to stay on-screen longer for intentional behavior design or to avoid unintended user dismissals.

Comment (lawrencewin) [reply apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts]: > Does this actually close the Menu? I just tried on the tester app and it didn't. Can you confirm?

Works on my machine, though I did add an escape press on the menu trigger button. I also keep the click on the defocus button where only pressing escape somehow breaks the e2etests and resetting focus somehow fixes that.

File: apps/fluent-tester/src/TestComponents/Menu/consts.ts (hunk 2 above)
Comment (ksiler) [reply apps/fluent-tester/src/TestComponents/Menu/consts.ts:20]: +1

**Review by ksiler — APPROVED** [review]
